### PR TITLE
(fix) O3-1895 Edit Past Visit button does not work

### PR DESCRIPTION
**Pull request by brandones**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary

Fixes the "edit past visit" functionality. Modifies the active visit tag so you can tell. Fixes active visit tag tooltip styling.

It's split into two commits, the change and a refactor. Review separately for pleasant diffs.

Note that this is not quite right—the Retrospective Entry tag should appear also after clicking "Start visit" and entering a visit in the past. But this requirement maybe needs to be specified a bit better. What if it's a multi-day visit started the previous day, which has not ended yet? Getting the logic right for identifying a retrospective visit is left to [O3-2242](https://issues.openmrs.org/browse/O3-2242).

## Screenshots

Peek 2023-07-07 23-38

Old:

New:

## Related Issue

https://issues.openmrs.org/browse/O3-1895

## Other

Depends on https://github.com/openmrs/openmrs-esm-core/pull/724

### CODE REVIEW

**Comment by brandones** [comment]
Made this a draft because I've been unable to test the routing update in [9a87890](https://github.com/openmrs/openmrs-esm-patient-chart/pull/1269/commits/9a87890870af5fa9bf0f08a6bd65bff6ad836c4b). Also I need to update this with respect to the core change.

File: packages/esm-patient-medications-app/src/order-basket/order-basket-search/drug-search.component.tsx
Comment (denniskigen) [inline packages/esm-patient-medications-app/src/order-basket/order-basket-search/drug-search.component.tsx:1]: Surprised how this got past ESLint

**Comment by brandones** [comment]
Anyone know why the build is broken?

```
 @openmrs/esm-form-entry-app:build: ./node_modules/@openmrs/esm-framework/src/internal.ts - Error: Module build failed (from ../../node_modules/@angular-devkit/build-angular/node_modules/@ngtools/webpack/src/ivy/index.js):
@openmrs/esm-form-entry-app:build: Error: /home/runner/work/openmrs-esm-patient-chart/openmrs-esm-patient-chart/packages/esm-form-entry-app/node_modules/@openmrs/esm-framework/src/internal.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
@openmrs/esm-form-entry-app:build:     at /home/runner/work/openmrs-esm-patient-chart/openmrs-esm-patient-chart/node_modules/@angular-devkit/build-angular/node_modules/@ngtools/webpack/src/ivy/loader.js:59:26
@openmrs/esm-form-entry-app:build: 
@openmrs/esm-form-entry-app:build: 
@openmrs/esm-form-entry-app:build: ERROR: command finished with error: command (/home/runner/work/openmrs-esm-patient-chart/openmrs-esm-patient-chart/packages/esm-form-entry-app) yarn run build exited (1)
```

**Comment by denniskigen** [comment]
> Anyone know why the build is broken?
> 
> ```
>  @openmrs/esm-form-entry-app:build: ./node_modules/@openmrs/esm-framework/src/internal.ts - Error: Module build failed (from ../../node_modules/@angular-devkit/build-angular/node_modules/@ngtools/webpack/src/ivy/index.js):
> @openmrs/esm-form-entry-app:build: Error: /home/runner/work/openmrs-esm-patient-chart/openmrs-esm-patient-chart/packages/esm-form-entry-app/node_modules/@openmrs/esm-framework/src/internal.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
> @openmrs/esm-form-entry-app:build:     at /home/runner/work/openmrs-esm-patient-chart/openmrs-esm-patient-chart/node_modules/@angular-devkit/build-angular/node_modules/@ngtools/webpack/src/ivy/loader.js:59:26
> @openmrs/esm-form-entry-app:build: 
> @openmrs/esm-form-entry-app:build: 
> @openmrs/esm-form-entry-app:build: ERROR: command finished with error: command (/home/runner/work/openmrs-esm-patient-chart/openmrs-esm-patient-chart/packages/esm-form-entry-app) yarn run build exited (1)
> ```

Could the `tsconfig.json` tweak be the culprit?

**Comment by brandones** [comment]
Thanks for the update @denniskigen ! And @ibacher , that looks like the old UI exactly. Are you sure you got the changes loaded?

I was going to try it again but I think I'm running into a bug in the extension renderer. Using `yarn start --sources packages/esm-patient-banner-app/`.

image

**Comment by brandones** [comment]
Fascinating! Okay well it does seem to have fixed it; updated core and now everything is loading nicely. The tooltip formatting looks correct, also.

image

**Comment by ibacher** [comment]
🤦 I just realised I got that result because I had forgotten to run the banner app locally. Sorry for the noise!

**Review by ibacher — APPROVED** [review]

**Review by denniskigen — APPROVED** [review]
Nice work, @brandones!

**Pull request merged**